modules/ROOT/pages/about-classloading-isolation.adoc

```diff
@@ -96,24 +94,58 @@ And the following `mule-artifact.json` descriptor file for the extension:
 
-image::simple-classloading-diagram.png[Simple classloading diagram]
+image::simple-classloading-diagram.png[Simple class-loading diagram]
 
 Artifact Class Loader::
-    A regular Java class loader pointing to the JAR files included in the extension.  This class loader will load all files and classes of the extension.
+    A regular Java class loader pointing to the JAR files included in the extension. This class loader loads all files and classes of the extension.
 
 Artifact Filtering Class Loader::
-    A wrapper created over the Artifact class loader, which will enforce the access restrictions to the extension code for foreign artifacts (the app or other plugins). It uses the content of the `mule-artifact.json` descriptor to determine what is public.
+    A wrapper created over the Artifact class loader, which enforces access restrictions to the extension code for foreign artifacts (the app or other plugins). It uses the content of the `mule-artifact.json` descriptor to determine what is public.
 
-Extension code::
-    The Mule extension code. It uses Artifact class loader (which does not have any restriction), and it is only able to locate resources of the plugin itself.
+Extension Code::
+    The Mule extension code. Uses the Artifact class loader (which does not have any restriction), and it is only able to locate resources of the plugin itself.
 
 Application Code::
     The Mule app code. It uses the Artifact Filtering class loader of the extension to prevent the app from accessing restricted code or resources.
 
+== Adding Dependencies to Connectors
+
+When the connectors in your application need access to a dependency in your project, configure the dependency to be visible by the connector's class loader. There are two different ways to do this:
+
+* Configure application dependencies as <<shared-libraries, shared libraries>>.
+* Configure <<additional-plugin-dependencies, additional plugin dependencies>> for the connector.
+
+[[shared-libraries]]
+=== Shared Libraries
+
+All dependencies (JAR files, for example) declared in the application's `pom.xml` file are visible to the application's class loader but not visible to the class loader of each connector used in the application. The Mule runtime engine class-loading mechanism isolates each connector to prevent it from accessing classes from other connectors. If a connector's class loader needs access to an application dependency, declare this dependency as a shared library.
+
+Consider an application that uses Anypoint Connector for Java, and the connector needs to use a class that is part of a JAR dependency declared in the application's `pom.xml` file. However, this is not possible, because the connector's class loader is not able to find that class. To make this class visible to the connector, you must declare the dependency that contains the class as a shared library in the Mule Maven plugin configuration of your application's `pom.xml` file.
```

Comment:
Anypoint Connector for Java (Java Connector) 
Same with the Database Connector below.